geode-core/src/integrationTest/java/org/apache/geode/cache/query/internal/ExecutionContextIntegrationTest.java

```diff
@@ -0,0 +1,340 @@
+
+    return i;
+  }
+
+  @Test
+  public void constructorShouldUseConfiguredMethodAuthorizer() {
+    ExecutionContext unsecuredContext = new QueryExecutionContext(null, server.getCache());
+    MethodInvocationAuthorizer noOpAuthorizer = unsecuredContext.getMethodInvocationAuthorizer();
+
+    // No security, no-op authorizer.
+    assertThat(noOpAuthorizer).isNotNull();
+    assertThat(noOpAuthorizer.getClass().getCanonicalName())
+        .startsWith("org.apache.geode.cache.query.internal.DefaultQueryService$$Lambda$");
+
+    server.stopMember();
+
+    // Security Enabled -> RestrictedMethodAuthorizer
+    server.withProperty(SECURITY_MANAGER, SimpleSecurityManager.class.getName())
+        .withProperty("security-username", "cluster")
+        .withProperty("security-password", "cluster")
+        .startServer();
+    ExecutionContext securedContext = new QueryExecutionContext(null, server.getCache());
+    MethodInvocationAuthorizer authorizer = securedContext.getMethodInvocationAuthorizer();
+    assertThat(authorizer).isNotNull();
+    assertThat(authorizer).isInstanceOf(RestrictedMethodAuthorizer.class);
+  }
+
+  @Test
+  public void testFunctionalAddToIndependentRuntimeItrMapWithoutIndex() throws Exception {
+    QCompiler compiler = new QCompiler();
+    List list = compiler.compileFromClause("/portfolio p, p.positions");
+    ExecutionContext context = new QueryExecutionContext(null, server.getCache());
+    context.newScope(context.associateScopeID());
+
+    for (Object o : list) {
+      CompiledIteratorDef iterDef = (CompiledIteratorDef) o;
+      @SuppressWarnings("unchecked")
+      Set<RuntimeIterator> dependencies = iterDef.computeDependencies(context);
+      context.addDependencies(new CompiledID("dummy"), dependencies);
+      RuntimeIterator rIter = iterDef.getRuntimeIterator(context);
+      context.bindIterator(rIter);
+      context.addToIndependentRuntimeItrMap(iterDef);
+
+      assertThat(rIter.getInternalId())
+          .as(" The index_internal_id is not set as per expectation of iter'n'")
+          .isEqualTo(rIter.getIndexInternalID());
+    }
+  }
+
+  @Test
+  public void testFunctionalAddToIndependentRuntimeItrMapWithIndex() throws Exception {
+    DefaultQueryService qs = new DefaultQueryService(server.getCache());
+    qs.createIndex("myindex", "pf.id", "/portfolio pf, pf.positions pos");
+    // compileFromClause returns a List<CompiledIteratorDef>
+    QCompiler compiler = new QCompiler();
+    List list = compiler.compileFromClause("/portfolio p, p.positions");
+    ExecutionContext context = new QueryExecutionContext(null, server.getCache());
+    context.newScope(context.associateScopeID());
+    Iterator iter = list.iterator();
+    int i = 0;
+    while (iter.hasNext()) {
+      CompiledIteratorDef iterDef = (CompiledIteratorDef) iter.next();
+      i = computeEvaluateAndAssertIterator(context, i, iterDef);
+    }
+  }
+
+  @Test
+  public void testObtainingRegionPath() throws Exception {
```

Comment:
Is this confirming that the region path in the context matches after computing dependencies or in this case computingUltimateDependencies?  (Not knowing what the difference is at this time...)